flows/download_catalog.py

```diff
@@ -294,62 +330,85 @@ def query_all(coo_centre, radius=24*u.arcmin, dist_cutoff=2*u.arcsec):
+	refcat = SkyCoord(ra=AT_results['ra'], dec=AT_results['decl'], unit=u.deg, frame='icrs')
 
-	ra_apass = np.array([r['ra'] for r in results_apass])
-	decl_apass = np.array([r['decl'] for r in results_apass])
-	apass = SkyCoord(ra=ra_apass, dec=decl_apass, unit=u.deg, frame='icrs')
+	#ra_apass = np.array([r['ra'] for r in results_apass])
+	#decl_apass = np.array([r['decl'] for r in results_apass])
+	apass = SkyCoord(ra=AT_apass['ra'], dec=AT_apass['decl'], unit=u.deg, frame='icrs')
 
 	# Match the two catalogs:
 	idx, d2d, _ = apass.match_to_catalog_sky(refcat)
+	sep_constraint = d2d <= dist_cutoff # Reject any match further away than the cutoff:
+	idx_apass = np.arange(len(idx)) # since idx maps apass to refcat
+
+	# Update results table with APASS bands of interest
+	AT_results['B_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['B_mag']
+	AT_results['V_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['V_mag']
+	AT_results['u_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['u_mag']
+
+	# Create SDSS cat
+	AT_sdss = query_sdss(coo_centre, radius=radius, dr=16, clean=True)
+	sdss = coords.SkyCoord(ra=AT_sdss['ra'], dec=AT_sdss['dec'], unit=u.deg, frame='icrs')
+
+	# Match to dist_cutoff sky distance (angular) apart
+	idx, d2d, _ = sdss.match_to_catalog_sky(refcat)
+	sep_constraint = d2d <= dist_cutoff
+	idx_sdss = np.arange(len(idx)) # since idx maps sdss to refcat
+	# TODO: Maybe don't (potentially) overwrite APASS uband with SDSS uband. Decide which is better.
+	AT_results['u_mag'][idx[sep_constraint]] = AT_sdss[idx_sdss[sep_constraint]]['psfMag_u']
+
+	# # Go through the matches and make sure they are valid:
+	# for k, i in enumerate(idx):
+	# 	# If APASS doesn't contain any new information anyway, skip it:
+	# 	if results_apass[k]['B_mag'] is None and results_apass[k]['V_mag'] is None \
+	# 		and results_apass[k]['u_mag'] is None:
+	# 		continue
+	#
+	# 	# Reject any match further away than the cutoff:
+	# 	if d2d[k] > dist_cutoff:
+	# 		continue
+	#
+	# 	# TODO: Use the overlapping magnitudes to make better match:
+	# 	#photdist = 0
+	# 	#for photfilt in ('g_mag', 'r_mag', 'i_mag', 'z_mag'):
+	# 	#	if results_apass[k][photfilt] and results[i][photfilt]:
+	# 	#		photdist += (results[i][photfilt] - results_apass[k][photfilt])**2
+	# 	#print( np.sqrt(photdist) )
+	#
+	# 	# Update the results "table" with the APASS filters:
+	# 	results[i].update({
+	# 		'V_mag': results_apass[k]['V_mag'],
+	# 		'B_mag': results_apass[k]['B_mag'],
+	# 		'u_mag': results_apass[k]['u_mag']
+	# 	})
+	#
+	# # Fill in empty fields where nothing was matched:
+	# for k in range(len(results)):
+	# 	if 'V_mag' not in results[k]:
+	# 		results[k].update({
+	# 			'B_mag': None,
+	# 			'V_mag': None,
+	# 			'u_mag': None
+	# 		})
+
+	# TODO: Adjust receiving functions so we can just pass the astropy table instead.
+	return [dict(zip(AT_results.colnames,row)) for row in AT_results.as_array()]
```

Comment:
Here we return back to list of dicts so we don't have to refactor the code for now.